# Overview

The interactive Variant Analysis \(IVA\) tool is an open-source project implemented to facilitate the filtering, analysis and interpretation of whole-genome variant data. This interactive tool allows the identification of genes affected by deleterious variants that segregate along with family pedigrees, case-control or sporadic samples.

IVA has been developed as part of the [OpenCGA](http://docs.opencb.org/display/opencga/Welcome+to+OpenCGA) project, making it easy to work with clinical and variant information stored in a Catalog and Variant storage instance.

### Main Features

* Allow loading VCF files and samples together with clinical data
* High-performance and scalable VCF and gVCF indexing
* VCF normalization and variant annotation
* Clinical interpretation analysis of samples and families

### Main Tools

* **Variant Browser:** It can efficiently browse annotated and indexed variant studies using a web-based application.
* **Clinical Interpretation:** It can select samples and families to perform an interpretation analysis, disease panels are also available.
* **Genome Browser:** A genome browser based on Genome Maps allows for an interactive and smooth variant visualisation.

{% hint style="success" %}
  **Rich Configuration** ðŸ’¡  

You can highly customise your IVA installation thanks to the rich configuration implemented. You can easily customise the navigation bar, welcome page and other options.
{% endhint %}











